s3.go

```diff
@@ -12,11 +12,13 @@ import (
 	"github.com/pkg/errors"
 )
 
+var session = func() S3Session {
+	cfg, _ := config.LoadDefaultConfig(context.TODO())
+	return s3.NewFromConfig(cfg)
+}()
```

Comment:
In this PR, I move session creation _out_ of the methods themselves and make the session into an interface. That way, I can swap out the interface with a mock to add tests that don't really hit S3.